airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/state/GlobalAsyncStateManager.java

```diff
@@ -169,13 +185,17 @@ public List<PartialAirbyteMessage> flushStates() {
 
           final var allRecordsCommitted = oldestStateCounter.get() == 0;
           if (allRecordsCommitted) {
-            output.add(oldestState.getLeft());
+            final PartialAirbyteMessage stateMessage = oldestState.getLeft();
+            final double flushedRecordsAssociatedWithState = stateIdToCounterForPopulatingDestinationStats.get(oldestStateId).doubleValue();
```

Comment:
why is this a double...? shouldn't it always be an integer

Reply:
So as discussed over call this is a double because the platform changes require it to be long